### PR TITLE
Resolve "Add SoftwareX paper to citation guidelines"

### LINKED ISSUE #1365: Add SoftwareX paper to citation guidelines
The [GwPy SoftwareX](https://www.sciencedirect.com/science/article/pii/S2352711021000029) paper should probably be added to the [citation guide page](https://gwpy.github.io/docs/stable/citing.html). Maybe also include the BibTeX for the paper:

```
@article{gwpy,
    title = "{GWpy: A Python package for gravitational-wave astrophysics}",
   author = {{Macleod}, D.~M. and {Areeda}, J.~S. and {Coughlin}, S.~B. and {Massinger}, T.~J. and {Urban}, A.~L.},
  journal = {SoftwareX},
   volume = {13},
    pages = {100657},
     year = {2021},
     issn = {2352-7110},
      doi = {10.1016/j.softx.2021.100657},
      url = {https://www.sciencedirect.com/science/article/pii/S2352711021000029},
}
```

### PR DESCRIPTION
**Pull request by mattpitkin**
I've made this PR to close #1365 in case it is wanted.

### CODE REVIEW

File: docs/citing.rst.in [hunk 1]
```diff
@@ -3,9 +3,83 @@ Citing GWpy
...
 
 If you have used GWpy as part of a project that leads to a scientific
-publication, please acknowledge this by citing the DOI for the version of
-GWpy that you have used.
+publication, please acknowledge this by citing the
+`GWpy paper <https://www.sciencedirect.com/science/article/pii/S2352711021000029>`_: D. M. Macleod *et al*, SoftwareX, **13**, 100657 (2021). The citation in various formats is given below:
 
+.. tabbed:: BibTeX
+
+   .. code-block::
```
Comment (duncanmmacleod) [inline docs/citing.rst.in:11]: Can you add fancy language highlighting?

```suggestion
   .. code-block:: bibtex
```

**Comment by duncanmmacleod** [comment]
@mattpitkin, this is a great contribution, thanks! I just need to check the rendering, then will merge.

**Review by areeda — APPROVED** [review]
LGTM

**Comment by duncanmmacleod** [comment]
@mattpitkin, can you rebase this please?

**Comment by duncanmmacleod** [comment]
@mattpitkin, please rebase against the upstream `main` branch, not `master.

**Comment by duncanmmacleod** [comment]
@mattpitkin, your rebase still has conflicts, please try again.

**Comment by duncanmmacleod** [comment]
This is failing because the `sphinx-panels` in conda-forge is out-of-date. Once conda-forge/sphinx-panels-feedstock#3 is merged and uploaded, we should be ready to rerun the CI here without change. #1394 has been proposed to improve the error reporting, but that wouldn't actually stop the jobs from failing.

**Comment by duncanmmacleod** [comment]
@mattpitkin, please rebase (again) and we should be able to get this merged today.

**Comment by duncanmmacleod** [comment]
@mattpitkin, I don't think this is an issue, but you seem to have merged, rather than rebased, so there are a bunch of extra commits in the history, compared to the single change you are proposing. In future you might find the following rebase command the easiest/cleanest way to update a feature branch:

```
git pull --rebase <origin> <source_branch>
```

e.g.

```
git pull --rebase upstream master
```

**Pull request merged**